sympy/solvers/solveset.py

```diff
@@ -843,6 +843,18 @@ def solveset(f, symbol=None, domain=S.Complexes):
     if not isinstance(f, (Expr, Number)):
         raise ValueError("%s is not a valid SymPy expression" % (f))
 
+    if isinstance(symbol, Function):
+        f_tup = f.args
+        for arg in f_tup:
+            if arg.has(symbol) or arg.is_Number:
+                continue
+            else:
+                raise ValueError("%s is not a valid expression" % (f))
```

Comment:
It doesn't mean that it is not a valid expression, rather it means that it is not solvable using this method and should be passed to other solvers.